lib/effect.fz

```diff
@@ -36,19 +36,24 @@ module:public effect (
   )
 is
 
+  # evaluate the current mode of the effect
+  #
   match r
-    _ effect_mode.plain   =>
-    i effect_mode.inst    => run i.f ()->unit
     _ effect_mode.repl    => replace
     _ effect_mode.default => default
-    _ effect_mode.new     =>
+    * =>
```

Comment:
Better keep all the single cases and not use `*`, this helps when adding a new modes to find all uses. Also, `effect_mode.inst` has a side effect, but it is unclear if this is still used anywhere.